freeipa/src/main/java/com/sequenceiq/freeipa/service/client/FreeipaClientTestService.java

```diff
@@ -0,0 +1,105 @@
+package com.sequenceiq.freeipa.service.client;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+import javax.ws.rs.InternalServerErrorException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.google.common.base.Joiner;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.freeipa.client.FreeIpaClient;
+import com.sequenceiq.freeipa.client.FreeIpaClientException;
+import com.sequenceiq.freeipa.client.model.Group;
+import com.sequenceiq.freeipa.client.model.User;
+import com.sequenceiq.freeipa.service.freeipa.FreeIpaClientFactory;
+
+@Service
+public class FreeipaClientTestService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FreeipaClientTestService.class);
+
+    @Inject
+    private FreeIpaClientFactory freeIpaClientFactory;
+
+    public String userShow(Long id, String name) {
+        FreeIpaClient freeIpaClient;
+        try {
+            freeIpaClient = freeIpaClientFactory.getFreeIpaClientForStackId(id);
+        } catch (Exception e) {
+            LOGGER.error("Error creating FreeIpaClient", e);
+            return "FAILED TO CREATE CLIENT";
+        }
+
+        try {
+            User user = freeIpaClient.userShow(name);
+            LOGGER.info("Groups: {}", user.getMemberOfGroup());
+            LOGGER.info("Success: {}", user);
+        } catch (Exception e) {
+            LOGGER.error("Error showing user {}", name, e);
+            return "FAILED TO SHOW USER";
+        }
+        return "END";
```

Comment:
I want to declare, I just moved this logic into the service, but I do not wanted to touch it, even if I am not agree with the desing of it

Reply:
it wasn't designed :D

Reply:
yes, it definitely seems it is not created for actual customer use cases :D